From 9b29201ce53b61cf40ead90b1fcadf7c4d8e71cd Mon Sep 17 00:00:00 2001
From: Noam Lessner <nlessner@habana.ai>
Date: Fri, 20 Mar 2020 11:43:15 +0200
Subject: [PATCH] [SW-10750] goya2: bypass secondary image efuse

In case we see that max allowed image timing doesn't allow secondary
image to load in time we can disable it via efuse

Change-Id: I9e5535fa51993cecf39a2530cfc4a62ac80df73b
Signed-off-by: Noam Lessner <nlessner@habana.ai>
---

diff --git a/app/arcboot_rom.c b/app/arcboot_rom.c
index 44465fa..1fd5be4 100644
--- a/app/arcboot_rom.c
+++ b/app/arcboot_rom.c
@@ -214,9 +214,12 @@
 
 	if (image_ptr == ARCBOOT_PRIMARY)
 		image_progress = PROGRESS_BRINGUP_PCI_PRI;
-	else
+	else {
 		image_progress = PROGRESS_BRINGUP_PCI_SEC;
-
+		if (arcboot_rom_global.config_fuse &
+		    EFUSE_SYSCFG_DISABLE_SECONDARY_IMG)
+			return ARCBOOT_BRINGUP_ERROR;
+	}
 	arcboot_rom_status_update_bringup(image_progress, PROGRESS_BRINGUP_START);
 	flash_image_offset = arcboot_flash_image_offset(image_ptr);
 	flash_ppboot_pcie_offset = (uint32_t)(FLASH_64KB_PADD +
@@ -270,7 +273,9 @@
 		arcboot_rom_status_update(PROGRESS_BRINGUP_PCI_PRI);
 		if (arcboot_rom_bringup_pcie_images(
 		    ARCBOOT_PRIMARY) != ARCBOOT_BRINGUP_OK) {
-			arcboot_rom_status_update(PROGRESS_BRINGUP_PCI_SEC);
+			if ((arcboot_rom_global.config_fuse &
+		      	    EFUSE_SYSCFG_DISABLE_SECONDARY_IMG) == 0)
+				arcboot_rom_status_update(PROGRESS_BRINGUP_PCI_SEC);
 			if (arcboot_rom_bringup_pcie_images(
 			    ARCBOOT_SECONDARY) != ARCBOOT_BRINGUP_OK) {
 				arcboot_rom_validation_fail();
diff --git a/common/efuse.h b/common/efuse.h
index 2d5141c..b31f353 100644
--- a/common/efuse.h
+++ b/common/efuse.h
@@ -17,6 +17,7 @@
 #define EFUSE_SYSCFG_OFFSET			0x3A
 #define EFUSE_SECURITY_FW_VERSION_OFFSET	0x3B
 
+#define EFUSE_SYSCFG_DISABLE_SECONDARY_IMG	BIT(21)
 #define EFUSE_SYSCFG_BACKUP_DISABLE		BIT(20)
 #define EFUSE_SYSCFG_DISABLE_TWI_IF_MASK	BIT(19)
 #define EFUSE_SYSCFG_PCIE_GEN_LIMIT		BIT(18)
